sdk/nodejs/tests/provider/experimental/testdata/array-types/index.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright 2025-2025, Pulumi Corporation.  All rights reserved.
+
+import * as pulumi from "@pulumi/pulumi";
+
+export interface MyComponentArgs {
+    anArrayOfStrings: Array<string>;
+    anArrayOfNumbers?: Array<number>;
+    anArrayOfBooleans: Array<boolean>;
+    inputArrayOfStrings: pulumi.Input<Array<string>>;
+    inputArrayOfNumbers?: pulumi.Input<Array<number>>;
+    inputArrayOfBooleans: pulumi.Input<Array<boolean>>;
+    anArrayOfInputStrings: Array<pulumi.Input<string>>;
+    anArrayOfInputNumbers?: Array<pulumi.Input<number>>;
+    anArrayOfInputBooleans: Array<pulumi.Input<boolean>>;
+    aListOfStrings: string[];
+    aListOfNumbers: number[];
+    aListOfBooleans?: boolean[];
```

Comment:
Add a test for our standard `pulumi.Input<pulumi.Input<T>[]>`? (both non plain?)